services/util/EchoInstantiation.hpp

```diff
@@ -27,6 +35,64 @@ namespace main_
         services::EchoOnMessageCommunication echo;
     };
 
+    template<std::size_t MessageSize>
+    struct EchoOnSesame
+    {
+        EchoOnSesame(hal::BufferedSerialCommunication& serialCommunication, services::MethodSerializerFactory& serializerFactory)
+            : cobs(serialCommunication)
+            , echo(windowed, serializerFactory)
+        {}
+
+        ~EchoOnSesame()
+        {
+            cobs.Stop();
+            windowed.Stop();
+        }
+
+        operator services::Echo&()
+        {
+            return echo;
+        }
+
+        void Reset()
+        {
+            echo.Reset();
+        }
+
+        services::SesameCobs::WithMaxMessageSize<MessageSize> cobs;
+        services::SesameWindowed windowed{ cobs };
+        services::EchoOnSesame echo;
+    };
+
+#if defined(EMIL_HAL_WINDOWS) || defined(EMIL_HAL_UNIX)
+    template<std::size_t MessageSize>
+    struct EchoOnUartBase
+    {
+        explicit EchoOnUartBase(infra::BoundedConstString portName)
+            : uart(infra::AsStdString(portName))
+        {}
+
+#if defined(EMIL_HAL_WINDOWS)
+        hal::UartWindows uart;
+#elif defined(EMIL_HAL_UNIX)
+        hal::UartUnix uart;
+#endif
```

Comment:
There should be a UartGeneric somewhere, you can make use of that instead of this

Reply:
Yes, there should, and I'll gladly review that pull request :)

Reply:
There already is one in hal/generic

Reply:
Oh. How did I miss that one?

Reply:
Replaced the classes